samples/goroutines_100k.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+)
+
+var nGoroutines int
+
+func memFunc() {
+	b := make([]int, 1024)
+	for i := 0; i < len(b); i++ {
+		b[0] = i
+		time.Sleep(1 * time.Microsecond)
+	}
+}
+
+func report() {
+	for {
+		n := runtime.NumGoroutine()
+		if n > 1 && nGoroutines != n {
+			nGoroutines = n
+			fmt.Printf("\r                                    ")
```

Comment:
At least I suggest replacing `\r` with `\n` - the output is strange at least in my editor terminal and it's worth while seeing those lines be printed.
s/100,000/100000 because the other ints are also printed without coma (had me confused at first).
Last I'd invert the logic a bit and 
```go
func report() {
	for n := runtime.NumGoroutine(); n > 1; {
		time.Sleep(20 * time.Millisecond)
		if n1 := runtime.NumGoroutine(); n != n1 {
			n = n1
			fmt.Printf("# Goroutines: %d/100000\n", n)
		}
	}
}
```